dlib/cuda/cudnn_dlibapi.cpp

```diff
@@ -799,7 +800,7 @@ namespace dlib
             // the cache.
             const auto cache_key = std::make_tuple(stride_y, stride_x, padding_y, padding_x, filters_nr, filters_nc);
             const auto iter = config_to_algo_cache.find(cache_key);
-            if (iter != config_to_algo_cache.end())
+            if (iter != config_to_algo_cache.end() && allow_cache_use == allow_cache_use::yes)
```

Comment:
This could be optimized by not doing the cache lookup in case cache use is not allowed in the first place.